client/spring-zeebe/src/main/java/io/camunda/zeebe/spring/client/connector/ConnectorConfiguration.java

```diff
@@ -15,9 +13,23 @@
 
 public class ConnectorConfiguration {
 
+  public static class OnMissingMetricsRecorder implements Condition {
+    @Override
+    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+      return context.getBeanFactory().getBeanNamesForType(MetricsRecorder.class).length<=0;
+    }
+  }
+
+  @Bean
+  @Conditional(value=OnMissingMetricsRecorder.class)
```

Comment:
Can we also simply do a [ConditionalOnMissingBean](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnMissingBean.html) here? As soon as someone creates a custom bean of the `MetricsRecorder` class, this here should be ignored. I think. 🙂

Reply:
We can't - as the current spring-zeebe module does not depend on Spring Boot - only Spring Context, hence the workaround. I think it should work for all relevant cases - but always happy to improve later on if we learn that it doesn't. We might also double check if somebody needs a pure spring (non spring boot) dependency - but I find it hard to find that out